src/EditorFeatures/Text/Extensions.SnapshotSourceText.cs

```diff
@@ -98,18 +111,11 @@ public override Encoding Encoding
             public ITextSnapshot TryFindEditorSnapshot()
                 => TryFindEditorSnapshot(this.TextImage);
 
-            protected static ITextBufferCloneService TextBufferFactory
+            protected ITextBufferCloneService TextBufferFactory
             {
                 get
                 {
-                    // simplest way to get text factory
-                    var ws = PrimaryWorkspace.Workspace;
-                    if (ws != null)
-                    {
-                        return ws.Services.GetService<ITextBufferCloneService>();
-                    }
-
-                    return null;
+                    return _workspace?.Services.GetService<ITextBufferCloneService>();
```

Comment:
i woudl feel better if this service was simply passed in in the start, rather than having to hold onto the _workspace itself.  is that doable?

Reply:
➡️ I replied to Jason's post regarding this with a proposed alternative